src/components/ReportActionItemIOUPreview.js

```diff
@@ -0,0 +1,162 @@
+
+    // The report currently being looked at
+    report: PropTypes.shape({
+
+        // IOU report ID associated with current report
+        iouReportID: PropTypes.number,
+
+        // Whether there is an outstanding amount in IOU
+        hasOutstandingIOU: PropTypes.bool,
+    }).isRequired,
+
+    /* Window Dimensions Props */
+    ...windowDimensionsPropTypes,
+
+    /* --- Onyx Props --- */
+    // Active IOU Report for current report
+    iou: PropTypes.shape({
+        // Email address of the manager in this iou report
+        managerEmail: PropTypes.string,
+
+        // Email address of the creator of this iou report
+        ownerEmail: PropTypes.string,
+
+        // Outstanding amount of this transaction
+        cachedTotal: PropTypes.string,
+    }),
+
+    // All of the personal details for everyone
+    personalDetails: PropTypes.objectOf(PropTypes.shape({
+
+        // This is either the user's full name, or their login if full name is an empty string
+        displayName: PropTypes.string.isRequired,
+    })).isRequired,
+
+    // Session info for the currently logged in user.
+    session: PropTypes.shape({
+        // Currently logged in user email
+        email: PropTypes.string,
+    }).isRequired,
+};
+
+const defaultProps = {
+    iou: {},
+};
+
+class ReportActionItemIOUPreview extends React.Component {
+    /**
+     * Creates IOU preview pay button.
+     *
+     * @returns {View}
+     */
+    createPayButton() {
+        return (
+            <TouchableOpacity
+                style={[styles.buttonSmall, styles.normalRadius, styles.buttonSuccess, styles.mt6]}
+            >
+                <Text
+                    style={[
+                        styles.buttonSmallText,
+                        styles.buttonSuccessText,
+                    ]}
+                >
+                    Pay
+                </Text>
+            </TouchableOpacity>
+        );
+    }
+
+    /**
+     * Creates IOU preview box view if outstanding amount is not 0.
+     *
+     * @returns {View}
+     */
+    createPreviewBox() {
+        const managerName = lodashGet(
+            this.props.personalDetails,
+            [this.props.iou.managerEmail, 'displayName'],
+            this.props.iou.managerEmail,
+        );
+        const ownerName = lodashGet(
+            this.props.personalDetails,
+            [this.props.iou.ownerEmail, 'displayName'],
+            this.props.iou.ownerEmail,
+        );
+        const managerAvatar = lodashGet(this.props.personalDetails, [this.props.iou.managerEmail, 'avatar'], '');
+        const ownerAvatar = lodashGet(this.props.personalDetails, [this.props.iou.ownerEmail, 'avatar'], '');
+        const sessionEmail = lodashGet(this.props.session, 'email', null);
+        const cachedTotal = this.props.iou.cachedTotal.replace(/[()]/g, '');
+
+        return (
+            <View style={[styles.iouPreviewBox, {width: this.props.isSmallScreenWidth ? '100%' : '50%'}]}>
```

Comment:
Could we have this use a width of 100%, but give it a max-width of something like `300` or something? This should simplify this statement here.